### PR TITLE
Updates test case template

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

* Improves readability of terms by making a dotted list
* Sorts the terms
* Corrects and adds links

## How to test this PR

Review only.

### CODE REVIEW

File: docs/internal-documentation/templates/specifications/tests/Template01.md [hunk 1]
```diff
@@ -233,28 +233,25 @@ No special terminology for this test case.
...
-The terms "in-bailiwick", "out-of-bailiwick" and "glue record" are defined in
-[RFC 8499][RFC 8499#section-7], section 7, pages 24-25. In this document, the
-term "in-bailiwick" is limited to the meaning "in domain" in
-[RFC 8499][RFC 8499#section-7]. The term "out-of-bailiwick" means what is not
-"in-bailiwick, in domain", in this document.
+* "DNS Lookup" - The term is used when a recursive lookup is used, though
+  any changes to the DNS tree introduced by an [undelegated test] must be
+  respected.
 
+* "in-bailiwick" - The terms are used as defined in [RFC 8499][RFC 8499#page-25],
```
Comment (mvw-afnic) [inline docs/internal-documentation/templates/specifications/tests/Template01.md:240]: “The term is used”, maybe?

**Review by mvw-afnic — CHANGES_REQUESTED** [review]
I see just a small typo.

File: docs/internal-documentation/templates/specifications/tests/Template01.md (hunk 1 above)
Comment (matsduf) [reply docs/internal-documentation/templates/specifications/tests/Template01.md]: Yes, fixed.

File: docs/internal-documentation/templates/specifications/tests/Template01.md [hunk 2]
```diff
@@ -237,7 +237,7 @@ No special terminology for this test case.
   any changes to the DNS tree introduced by an [undelegated test] must be
   respected.
 
-* "in-bailiwick" - The terms are used as defined in [RFC 8499][RFC 8499#page-25],
+* "in-bailiwick" - The terms is used as defined in [RFC 8499][RFC 8499#page-25],
```
Comment (mvw-afnic) [inline docs/internal-documentation/templates/specifications/tests/Template01.md:240]: There’s also “terms”, that needs to be changed to “term”.

**Review by mvw-afnic — APPROVED** [review]
Looks good to me.

**Review by tgreenx — APPROVED** [review]

**Pull request merged**